crates/devices/virtio-vsock/src/packet.rs

```diff
@@ -460,19 +460,30 @@ impl<'a, B: BitmapSlice> VsockPacket<'a, B> {
             return Err(Error::InvalidHeaderLen(pkt.len()));
         }
 
-        let data_desc = desc_chain.next().ok_or(Error::DescriptorChainTooShort)?;
+        // Starting from Linux 6.2 the virtio-vsock driver can use a single descriptor for both
+        // header and data.
+        let data_slice = if chain_head.len() >= PKT_HEADER_SIZE as u32 + pkt.len() {
```

Comment:
The addition operation can overflow. Unfortunately not having a maximum (and reasonable) max packet size means that a wrongly initialized value can lead to an overflow here in case you're also dealing with a malicious user.

Reply:
To reproduce:
```rust
        // On the TX path, it is allowed to not have a data descriptor.
        let v = vec![Descriptor::new(0x10_0000, PKT_HEADER_SIZE as u32, 0, 0)];
        let mut chain = queue.build_desc_chain(&v[..1]).unwrap();

        let header = PacketHeader {
            src_cid: SRC_CID.into(),
            dst_cid: DST_CID.into(),
            src_port: SRC_PORT.into(),
            dst_port: DST_PORT.into(),
            len: u32::MAX.into(),
            type_: 0.into(),
            op: 0.into(),
            flags: 0.into(),
            buf_alloc: 0.into(),
            fwd_cnt: 0.into(),
        };
        mem.write_obj(header, GuestAddress(0x10_0000)).unwrap();

        let packet = VsockPacket::from_tx_virtq_chain(&mem, &mut chain, u32::MAX).unwrap();
```

Reply:
> The addition operation can overflow. Unfortunately not having a maximum (and reasonable) max packet size means that a wrongly initialized value can lead to an overflow here in case you're also dealing with a malicious user.

Which reminds me that I should at some point resume work (addressing comments actually) on https://lists.oasis-open.org/archives/virtio-comment/202206/msg00030.html 😄 .

Reply:
Okay, I'll change the check to `if chain_head.len() - PKT_HEADER_SIZE as u32 >= pkt.len()`, so every potential overflow will go to the `else` branch and handled as before this patch.

Is that okay for you?

Reply:
I think this will still make the fuzzer fail because overflows are panics in debug mode, and if I am not mistaken you can still have an underflow? Will double check.

Reply:
At that point, we already checked that `chain_head.len()` is  `>= PKT_HEADER_SIZE`, so I think we can't have underflow or overflow. Or am I forgetting something? (My mind is already in the holiday mood 🎄).

Reply:
Yeah, I know what you mean, same here 🤶. Let me take a better look after some more ☕

Reply:
This overflow is indeed no longer possible. Resolving this conversation.